src/poetry/core/masonry/api.py

```diff
@@ -36,10 +40,21 @@ def get_requires_for_build_wheel(
 get_requires_for_build_sdist = get_requires_for_build_wheel
 
 
+def _apply_local_version_label(poetry: Poetry, config_settings: dict[str, Any]) -> None:
+    """Apply local version label from config settings to the poetry package version if present."""
+    if local_version_label := config_settings.get("local-version"):
+        poetry.package.version = poetry.package.version.replace(
+            local=local_version_label
+        )
```

Comment:
```suggestion
def create_poetry(config_settings: dict[str, Any] | None = None) -> Poetry:
    config_settings = config_settings or {}
    poetry = Factory().create_poetry(Path().resolve(), with_groups=False)
    poetry.package.version = poetry.package.version.replace(
        local=config_settings.get("local-version")
    )
    return poetry

```

Maybe something like this so we can reduce some duplicated code?

Reply:
I'd go as far as delaying `Poetry` object creation until inside the builder, passing `config_settings` and directory info to builder constructor. Then inside the builder we can create `Poetry` with proper version (preferably with some kind of `Poetry.with_version()` method or something.

Reply:
That might be a bigger change than necessary right now, would be good to keep the config settings contained in `masonry.api` for now.

Reply:
My goal is to avoid adding anything to the `masonry.api` scope. That module should only contain the build backend methods (at least in my book).

Reply:
Passing the `config_settings` to the builder sounds good :+1: 

I wouldn't let the builder create the `Poetry` object, because the code now reads like "build for this poetry package". Which is absolute what I'd expect. Otherwise it would read like "build whatever is in this directory", which is not that clear.